### PR TITLE
Add pkg-config file

### PR DESCRIPTION
**Pull request by Blarse**
Hi,

I'm adding passwdqc support to systemd and it would be very nice to have pkg-config file so that meson can find the dependency.

### CODE REVIEW

File: passwdqc.pc.in [hunk 1]
```diff
@@ -0,0 +1,10 @@
+prefix=@PREFIX@
+exec_prefix=${prefix}
+includedir=@INCLUDEDIR@
+libdir=@LIBDIR@
+
```
Comment (ldv-alt) [inline passwdqc.pc.in:5]: Are these lines needed at all?

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,7 @@
 *.o
 *.so
 *.so.*
+passwdqc.pc
```
Comment (ldv-alt) [inline .gitignore:4]: This should rather be `/passwdqc.pc`.

File: Makefile [hunk 3]
```diff
@@ -153,7 +157,7 @@ utils_wrapped: $(BINS)
 $(SHARED_LIB): $(OBJS_LIB) $(MAP_LIB)
 	$(LD_lib) $(LDFLAGS_lib) $(OBJS_LIB) $(LDLIBS_lib) -o $(SHARED_LIB)
 
-$(DEVEL_LIB): $(SHARED_LIB)
+$(DEVEL_LIB): $(SHARED_LIB) $(PKGCONFIG)
```
Comment (ldv-alt) [inline Makefile:160]: Does `$(DEVEL_LIB)` really depend on `$(PKGCONFIG)`?

File: Makefile [hunk 4]
```diff
@@ -180,6 +184,13 @@ pwqfilter.o: pwqfilter.c passwdqc_filter.h passwdqc.h
 .c.o:
 	$(CC) $(CPPFLAGS_lib) $(CFLAGS_lib) -c $*.c
 
+$(PKGCONFIG):
+	sed -e "s|@PREFIX@|/usr|g" \
+	    -e "s|@INCLUDEDIR@|$(INCLUDEDIR)|g" \
+	    -e "s|@LIBDIR@|$(DEVEL_LIBDIR)|g" \
```
Comment (ldv-alt) [inline Makefile:190]: This might be redundant, see below.

**Review by ldv-alt — CHANGES_REQUESTED** [review]

File: passwdqc.pc.in (hunk 1 above)
Comment (Blarse) [reply passwdqc.pc.in]: You're right these lines are needless

File: .gitignore (hunk 2 above)
Comment (Blarse) [reply .gitignore]: ok

File: Makefile (hunk 3 above)
Comment (Blarse) [reply Makefile:160]: Well probably not directly but in my opinion passwdqc.pc should be generated each time dev library is created. Is there a better way to do that?

**Comment by ldv-alt** [comment]
Please also add `$(PKGCONFIG)` to `PROJ`.

File: Makefile [hunk 5]
```diff
@@ -180,6 +184,9 @@ pwqfilter.o: pwqfilter.c passwdqc_filter.h passwdqc.h
 .c.o:
 	$(CC) $(CPPFLAGS_lib) $(CFLAGS_lib) -c $*.c
 
+$(PKGCONFIG):
+	sed -e "s|@VERSION@|$(VERSION)|g" $@.in > $@
```
Comment (ldv-alt) [inline Makefile:188]: Also, $(PKGCONFIG) should depend on its template.

File: Makefile (hunk 3 above)
Comment (ldv-alt) [reply Makefile:160]: Unlike `$(DEVEL_LIB)`, `$(PKGCONFIG)` is not needed to build anything else.
Maybe just add `$(PKGCONFIG)` to `all_wrapped`?

Comment (Blarse) [reply Makefile:160]: > `$(PKGCONFIG)` is not needed to build anything else.

That make point, will add it to `all_wrapped`

File: Makefile (hunk 5 above)
Comment (Blarse) [reply Makefile]: done

File: Makefile [hunk 6]
```diff
@@ -18,6 +19,8 @@ MAP_PAM = pam_passwdqc.map
 SHLIBMODE = 755
 HEADER = passwdqc.h
 INCMODE = 644
+PKGCONFIG = passwdqc.pc
+PKGCONFMOD = 644
```
Comment (ldv-alt) [inline Makefile:23]: The last bit: for consistency, please rename `PKGCONFMOD` to `PKGCONFMODE`.

**Review by ldv-alt — APPROVED** [review]
This looks OK now.
@solardiz, please approve CI for this PR.

**Review by solardiz — APPROVED** [review]
Approving based on @ldv-alt's review (thanks!)

I think this addition should be mentioned in `CHANGES`. Can this be added, please? New section "Significant changes since 2.0.2.", which we'll then rename shortly before release.

**Comment by ldv-alt** [comment]
> Approving based on @ldv-alt's review (thanks!)

There is `1 workflow awaiting approval`, it needs an explicit approval to run CI tests, to make sure they pass (they do).

**Comment by Blarse** [comment]
> I think this addition should be mentioned in `CHANGES`. Can this be added, please? New section "Significant changes since 2.0.2.", which we'll then rename shortly before release.

done

**Comment by solardiz** [comment]
> There is `1 workflow awaiting approval`, it needs an explicit approval to run CI tests, to make sure they pass (they do).

Ah, right. Approved it. (I wonder why it didn't run automatically?)

Everything looks good to me now, so I'll merge.

**Pull request merged**